apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberPreferenceRepository, SubscriberPreferenceEntity } from '@novu/dal';
+import { UpdateSubscriberPreferenceCommand } from './update-subscriber-preference.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class UpdateSubscriberPreference {
+  constructor(private subscriberPreferenceRepository: SubscriberPreferenceRepository) {}
+
+  async execute(command: UpdateSubscriberPreferenceCommand): Promise<SubscriberPreferenceEntity> {
+    const userPreference = await this.subscriberPreferenceRepository.findOne({
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+      _subscriberId: command.subscriberId,
+      _templateId: command.templateId,
+    });
+
+    if (!userPreference) {
+      const channelObj = {} as Record<'email' | 'sms' | 'in_app' | 'direct' | 'push', boolean>;
+      channelObj[command.channel?.type] = command.channel?.enabled;
+
+      return await this.subscriberPreferenceRepository.create({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: command.subscriberId,
+        _templateId: command.templateId,
+        enabled: command.enabled !== false,
+        channels: channelObj.hasOwnProperty('undefined') ? null : channelObj,
+      });
+    }
+
+    const updatePayload: Partial<SubscriberPreferenceEntity> = {};
+
+    if (command.enabled || command.enabled === false) {
```

Comment:
```suggestion
    if (command.enabled != null) {
```
This should help with null and undefined